src/bin/pgcopydb/pgcmd.c

```diff
@@ -504,25 +535,13 @@ pg_dump_db_extension_namespace_hook(void *ctx, SourceExtension *ext)
 	DumpExtensionNamespaceContext *context =
 		(DumpExtensionNamespaceContext *) ctx;
 
-	char **args = context->args;
+	char **extNamespaces = context->extNamespaces;
 
 	char *nspname = ext->extnamespace;
 
 	if (!streq(nspname, "public") && !streq(nspname, "pg_catalog"))
 	{
-		/* check that we still have room for --exclude-schema args */
-		if (PG_CMD_MAX_ARG < (*(context->argsIndex) + 2))
-		{
-			log_error("Failed to call pg_dump, "
-					  "too many schema are excluded: "
-					  "argsIndex %d > %d",
-					  *(context->argsIndex) + 2,
-					  PG_CMD_MAX_ARG);
-			return false;
-		}
-
-		args[(*context->argsIndex)++] = "--exclude-schema";
-		args[(*context->argsIndex)++] = nspname;
+		extNamespaces[(*context->extNamespaceCount)++] = strdup(nspname);
```

Comment:
This fixes the invalid memory access.

Reply:
Yeah but it's still missing a buffer overflow check before adding to the new array, right?